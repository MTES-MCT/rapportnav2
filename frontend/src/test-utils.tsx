import React, { ReactElement } from 'react'
import { render, renderHook, RenderOptions } from '@testing-library/react'
import UIThemeWrapper from './ui/ui-theme-wrapper'
import { BrowserRouter } from 'react-router-dom'

const AllTheProviders = ({ children }: { children: React.ReactNode }) => {
  return (
    <UIThemeWrapper>
      <BrowserRouter>{children}</BrowserRouter>
    </UIThemeWrapper>
  )
}

const customRender = (ui: ReactElement, options?: Omit<RenderOptions, 'wrapper'>) =>
  render(ui, { wrapper: AllTheProviders, ...options })

const customRenderHook = (hook: any, options?: Omit<RenderOptions, 'wrapper'>) =>
  renderHook(() => hook(), { wrapper: AllTheProviders, ...options })

export * from '@testing-library/react'
export { customRender as render }
export { customRenderHook as renderHook }
