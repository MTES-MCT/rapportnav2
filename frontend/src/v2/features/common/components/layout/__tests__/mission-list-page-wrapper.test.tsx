import { render } from '../../../../../../test-utils'
import MissionListPageWrapper from '../mission-list-page-wrapper'

describe('MissionListPageWrapper', () => {
  it('should match the snapshot', () => {
    const wrapper = render(
      <MissionListPageWrapper
        footer={<>Footer</>}
        header={<>Header</>}
        sidebar={<>SideBar</>}
        children={<>Main Page</>}
      />
    )
    expect(wrapper).toMatchSnapshot()
  })
})
