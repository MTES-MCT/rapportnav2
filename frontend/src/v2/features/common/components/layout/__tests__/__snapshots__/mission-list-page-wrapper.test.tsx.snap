// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MissionListPageWrapper > should match the snapshot 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="rs-container"
        style="min-height: 100vh; max-height: 100vh; overflow: hidden;"
      >
        <div
          class="rs-stack"
          style="align-items: center; flex-direction: column; width: 100%; height: 100vh;"
        >
          <div
            class="rs-stack-item"
            style="width: 100%;"
          >
            Header
          </div>
          <div
            class="rs-stack-item"
            style="flex: 1; width: 100%;"
          >
            <section
              class="rs-container"
              style="height: 100%;"
            >
              <main
                class="rs-content"
                style="max-height: calc(100vh - 104px - 50px); overflow: hidden; display: flex;"
              >
                Main Page
              </main>
            </section>
          </div>
          <div
            class="rs-stack-item"
            style="width: 100%; min-height: 50px; border-top: 2px solid #CCCFD6;"
          >
            Footer
          </div>
        </div>
      </section>
    </div>
    <div
      class="rs-toast-provider"
    >
      <div
        class="rs-toast-container rs-toast-container-top-center"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-center"
      />
      <div
        class="rs-toast-container rs-toast-container-top-start"
      />
      <div
        class="rs-toast-container rs-toast-container-top-end"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-start"
      />
      <div
        class="rs-toast-container rs-toast-container-bottom-end"
      />
    </div>
  </body>,
  "container": <div>
    <section
      class="rs-container"
      style="min-height: 100vh; max-height: 100vh; overflow: hidden;"
    >
      <div
        class="rs-stack"
        style="align-items: center; flex-direction: column; width: 100%; height: 100vh;"
      >
        <div
          class="rs-stack-item"
          style="width: 100%;"
        >
          Header
        </div>
        <div
          class="rs-stack-item"
          style="flex: 1; width: 100%;"
        >
          <section
            class="rs-container"
            style="height: 100%;"
          >
            <main
              class="rs-content"
              style="max-height: calc(100vh - 104px - 50px); overflow: hidden; display: flex;"
            >
              Main Page
            </main>
          </section>
        </div>
        <div
          class="rs-stack-item"
          style="width: 100%; min-height: 50px; border-top: 2px solid #CCCFD6;"
        >
          Footer
        </div>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
